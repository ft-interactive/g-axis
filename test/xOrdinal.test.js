/**
 * @file
 * Test suite for xOrdinal
 */
import { toMatchImageSnapshot } from 'jest-image-snapshot';

jest.setTimeout(20000);
expect.extend({ toMatchImageSnapshot });

beforeAll(global.build('xOrdinal'));
beforeEach(global.start);

test('bottom-aligned, default scales', async () => {
    await global.page.evaluate(async () => {
        const sharedConfig = {
            source: 'g-axis',
            subtitle: 'Bottom-aligned, default scales',
            title: 'xOrdinal test',
        };

        const svg = await window.d3.select(document.querySelector('svg'));
        const currentFrame = window.gChartframe.webFrameMDefault(sharedConfig);

        // Set up the chart frame
        svg.call(currentFrame);

        // Instantiate xAxis
        const xAxis = window.xOrdinal()
            .plotDim([currentFrame.dimension().width, currentFrame.dimension().height])
            .rangeRound([0, currentFrame.dimension().width])
            .align('bottom')
            .rem(currentFrame.rem())
            .frameName('webFrameMDefault');

        // Set up xAxis
        currentFrame.plot().call(xAxis);

        // Translate to bottom of plot
        xAxis.xLabel()
            .attr('transform',
                `translate(0,${currentFrame.dimension().height - (currentFrame.rem() / 1.5)})`);
    });

    const image = await global.page.screenshot();
    expect(image).toMatchImageSnapshot();
});

test('top-aligned, default scales', async () => {
    await global.page.evaluate(async () => {
        const sharedConfig = {
            source: 'g-axis',
            subtitle: 'Top-aligned, default scales',
            title: 'xOrdinal test',
        };

        const svg = await window.d3.select(document.querySelector('svg'));
        const currentFrame = window.gChartframe.webFrameMDefault(sharedConfig);

        // Set up the chart frame
        svg.call(currentFrame);

        // Instantiate xAxis
        const xAxis = window.xOrdinal()
            .plotDim([currentFrame.dimension().width, currentFrame.dimension().height])
            .rem(currentFrame.rem())
            .rangeRound([0, currentFrame.dimension().width])
            .align('top')
            .frameName('webFrameMDefault');

        // Set up xAxis
        currentFrame.plot().call(xAxis);
    });

    const image = await global.page.screenshot();
    expect(image).toMatchImageSnapshot();
});
