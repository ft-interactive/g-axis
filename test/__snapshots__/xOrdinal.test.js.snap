// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bottom-aligned, default scales 1`] = `
"<g class=\\"highlights\\"></g>
<g class=\\"axis xAxis\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"middle\\" transform=\\"transform(0, 263)\\">
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(78, 0)\\">
    <line stroke=\\"#000\\" y2=\\"10\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"13\\" dy=\\"0.71em\\" id=\\"webFrameMDefaultxLabel\\">Oranges</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(251, 0)\\">
    <line stroke=\\"#000\\" y2=\\"10\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"13\\" dy=\\"0.71em\\" id=\\"webFrameMDefaultxLabel\\">Lemons</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(424, 0)\\">
    <line stroke=\\"#000\\" y2=\\"10\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"13\\" dy=\\"0.71em\\" id=\\"webFrameMDefaultxLabel\\">Apples</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(597, 0)\\">
    <line stroke=\\"#000\\" y2=\\"10\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"13\\" dy=\\"0.71em\\" id=\\"webFrameMDefaultxLabel\\">Pears</text>
  </g>
</g>"
`;

exports[`top-aligned, default scales 1`] = `
"<g class=\\"highlights\\"></g>
<g class=\\"axis xAxis\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"middle\\">
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(78, 0)\\">
    <line stroke=\\"#000\\" y2=\\"-10\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"-13\\" dy=\\"0em\\" id=\\"webFrameMDefaultxLabel\\">Oranges</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(251, 0)\\">
    <line stroke=\\"#000\\" y2=\\"-10\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"-13\\" dy=\\"0em\\" id=\\"webFrameMDefaultxLabel\\">Lemons</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(424, 0)\\">
    <line stroke=\\"#000\\" y2=\\"-10\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"-13\\" dy=\\"0em\\" id=\\"webFrameMDefaultxLabel\\">Apples</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(597, 0)\\">
    <line stroke=\\"#000\\" y2=\\"-10\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"-13\\" dy=\\"0em\\" id=\\"webFrameMDefaultxLabel\\">Pears</text>
  </g>
</g>"
`;
